packages/api/src/module/write/shared/application/application-command.factory.ts

```diff
@@ -28,14 +28,17 @@ export class ApplicationCommandFactory {
     const generateId = () => this.idGenerator.generate();
     const currentTime = () => this.timeProvider.currentTime();
 
+    const id = generateId();
```

Comment:
What is the purpose of this change? 

Good articles about correlationId and causationId:
https://blog.arkency.com/correlation-id-and-causation-id-in-evented-systems/
https://railseventstore.org/docs/v2/correlation_causation/

Reply:
I did it because of order of generating IDs. 
Before: 
In handler was generated ID, then was generated command Id and then correlationID
Now: 
ID generated in handler is at the end 

so it test lastGeneratedId() returns Id created in handler, not correlationID